File: Unidad 2-Estructuras de Control/Ejercicio48.c
#include <stdio.h>
/* Programa que lee el suldo base y antiguedad del empleado
# Calcula el incentivo con base a ello
# e imprime el sueldo base, incentivo y sueldo a pagar
#  Audny D. Correa Ceballos (Equipo: 'about:blank') */

//PROTOTIPOS DE LAS FUNCIONES
void leerEntradas(int* , int*);
int calcularIncentivo(int, int );
int calcularSueldoAPagar (int , int );
void imprimirSalidas(int, int, int);

int main(){
	//Definir las varianles
	int sueldoBase, antiguedad, incentivoGenerado, sueldoAPagar;
	//Leer entradas
	leerEntradas(&sueldoBase, &antiguedad);
	//Proceso
	incentivoGenerado= calcularIncentivo(sueldoBase, antiguedad);
	sueldoAPagar = calcularSueldoAPagar (sueldoBase, incentivoGenerado);
	//Salidas
	imprimirSalidas(sueldoBase, incentivoGenerado, sueldoAPagar);
	
	return 0;
}

void leerEntradas(int*_sueldoBase, int*_antiguedad){
	scanf("%d", "%d", _sueldoBase, _antiguedad);
}

int calcularIncentivo(int _sueldoBase, int _antiguedad){
	int incentivo=0;
	if (_antiguedad >= 1 && _antiguedad <= 3);{
		incentivo = _sueldoBase * 0.01;
	}if (_antiguedad >= 4 && _antiguedad <= 6);{
		incentivo = _sueldoBase * 0.03;
	}if (_antiguedad >= 7 && _antiguedad <= 9);{
		incentivo = _sueldoBase * 0.05;
	}if (_antiguedad >= 10);{
		incentivo = _sueldoBase * 0.07;
	}
	return incentivo;
}

int calcularSueldoAPagar (int _sueldoBase, int _incentivoGenerado){
	int sueldoAFuturo;
	sueldoAFuturo = _sueldoBase + _incentivoGenerado;
	return sueldoAFuturo;
}

void imprimirSalidas(int _sueldoBase, int _incentivoGenerado, int _sueldoAPagar){
	printf("%d", "%d", "%d", _sueldoBase, _incentivoGenerado, _sueldoAPagar);
}

File: Unidad 2-Estructuras de Control/Ejercicio48.py
#Ejercicio #48
# Programa que lee el suldo base y antiguedad del empleado
# Calcula el incentivo con base a ello
# e imprime el sueldo base, incentivo y sueldo a pagar
#  Audny D. Correa Ceballos 

#Proceso: Definimos una funcion que servira para calcular el incentivo 
#de acuerdo a las entradas que de el usuario
def main(sueldoBase, antiguedad):
    #inicializar el incentivo en 0
    incentivo = 0
    if (antiguedad >= 1 and antiguedad <= 3):
        incentivo = sueldoBase * 0.01
    elif (antiguedad >= 4 and antiguedad <= 6):
        incentivo = sueldoBase * 0.03
    elif (antiguedad >= 7 and antiguedad <= 9):
        incentivo = sueldoBase * 0.05
    elif (antiguedad >= 10):
        incentivo = sueldoBase * 0.07

    return(incentivo)

#ENTRADAS: Pedimos el salario base y la antiguedad del usuario
sueldoBase = int(input())
antiguedad = int(input())

#Es un subproceso que calcula el salario a pagar con la suma del salario inicial 
# y de acuerdo al incentivo que se obtuvo con la funcion main
sueldoaPagar = main(sueldoBase, antiguedad) + sueldoBase

#SALIDAS: 
#El sueldo base que el usuario ingreso el principio de la funcion
print("Sueldo Base:", sueldoBase)
#El incentivo que se calcul'o en la funcion main
print("Incentivo:", main(sueldoBase, antiguedad))
#Sueldo a pagar 
print("Sueldo a pagar:", sueldoaPagar)